src/lib/utils/SynapseClient.ts

```diff
@@ -496,15 +499,15 @@ export const getAsyncResultFromJobId = async <TRequest, TResponse>(
       undefined,
       BackendDestinationEnum.REPO_ENDPOINT,
     )
-    setCurrentAsyncResult?.(response)
+    setCurrentAsyncStatus?.(response)
   }
 
   if (response.jobState === 'FAILED') {
     /**
      * While we technically already have the failure reason in the response, the HTTP response doesn't give a helpful error code (e.g. 403)
      * that we can use for an error banner. We can get the HTTP code if we fetch the response body directly.
      */
-    doGet<TResponse>(
+    await doGet<TResponse>(
```

Comment:
This `await` is critical for capturing the error; the failed response was returned before this request would throw an error.